tests/test-cases/Group1-Docker-Commands/1-38-Docker-Exec.md

```diff
@@ -25,3 +25,66 @@ This test requires that a vSphere server is running and available
+2. Create a container running `sleep` to simulate a bounded time process.
+3. Run the container and detach from it.
+4. Start 5 simple exec operations(/bin/ls) in parallel against the detached container.
+5. Wait for each exec to finish and check the rc and output for correctness
+6. Wait for the container to finish the sleep and exit.
+7. Check container run for correctness
+
+## Expected Outcome
+* step 1 Should successfully complete with an rc of 0.
+* step 2 Should successfully complete with an rc of 0.
+* step 3 Should successfully launch the container and detach from it. The container should remain running for 5 seconds only.
+* step 4 All 5 execs should be started successfully, we expect all to succeed.
+* step 5 All execs should have an rc of 0 and the correct root directories of the stashed busybox file system.
+* step 6 The container should exit.
+* step 7 The container should have an exit code of 0.
+
+# Exec Power Off test for long running Process
+## Purpose:
+This test is designed to test running exec's against a container running a process which would not naturally exit(long running). Then after running many execs concurrently we explicitly stop the container. We should then handle the execs in their varying states properly.
+
+## Test Steps
+1. Pull an image that contains `/bin/top`. Busybox suffices here.
+2. Create a container running `/bin/top` to simulate a long running process that should not exit on it's own.
+3. Run the container and detach from it.
+4. Start 20 simple exec operations in parallel against the detached container.
+5. Explicitly stop the container while the execs are still running in order to trigger the exec power off errors.
+6. collect all output from the parallel exec operations.
+
+## Expected Outcome
+* step 1 should successfully complete with an rc of 0
+* step 2 should successfully complete with an rc of 0
+* step 3 should successfully launch the container and detach from it. The container should remain running for 5 seconds only.
+* step 4 all 20 execs should be started successfully, we expect most if not all to fail.
+* step 5 container should halt successfully.
+* step 6 should contain the error message for exec operations that are interrupted by a power off operation. Specifically a poweroff that was explicitly triggered.
+
+# Exec Power Off test for short Running Process
+## Purpose:
+This test is designed to start a container and then initiate a long running exec operation(/bin/top). The expectation is that the exec will succeed in execution and run for the length of the containers life. Once the container exits the exec operation should also exit cleanly without error.
+
+## Test Steps
+1. Pull an image that contains `sleep`. Busybox suffices here.
+2. Create a container running `sleep` to simulate a short running process.
+3. Run the container and detach from it.
+4. Start 20 simple exec operations in parallel against the detached container.
```

Comment:
this is not the steps you have implemented below

Reply:
ah yeah I need to clean that up since I changed the test. :( fixing that now!